fix for starting services with a "run" method

ericblade opened the pull request:
Previously having a "run" method in a service that is not run via the
unified server would generate a runtime error that "name" does not exist.

This fixes it, so you can use services that are intended to start via
"run()", for nice things like setting up dependency injection

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/webosose/mojoservicelauncher?pullRequest=1) <br/>All committers have signed the CLA.

yogishSan reviewed (commented):
Dear Eric,
Is this verified with use case scenarios  for node services?

ericblade commented:
@yogishSan yeah, this doesn't change any functionality, only fixes a piece that has been broken for a very long time (but was fixed for the unified service case)

right now, if you build a service module and export a run() method from it, you'll just get an error on console, that the 'name' variable isn't defined, and the service won't do anything.  After this change, it will work.  

index.js:
````
class MyServiceClass {
....
};
function run() {
    new MyServiceClass();
}
module.exports = run;
````

should work, but does not, without this patch

It changes no behavior for services that don't export a run function from their main module, so no current services are using that feature.

I have a use case involving automated testing, that would be *very* good to have that fixed, otherwise I have to resort to workarounds such as using timeouts or global variables to flag that a unit is or isn't being tested.

ericblade commented:
thanks for approving, how long to merge? :-)

flowergom commented:
Hi @ericblade.
Your PR will be merged by our internal processes. Not merged on github directly. 
Therefore, it will take some time for actual release.

ericblade commented:
so... PRs will just stay open forever on github, and no one will ever know if/when they are handled?  Is that why there are PRs in some of the repos here dating back 7 months, with no apparent action?  Seems like a process that could really stand to have some improvement.

ericblade committed:
support node ES modules

- as NODE_PATH is ignored for ESM import statements, symlink webos-service into the service node_modules

ericblade commented:
This was eventually accepted, thank you.
